lib/util/util.CombiningMessageProvider.js

```diff
@@ -0,0 +1,53 @@
+this.util = this.util || {};
+
+util.CombiningMessageProvider = ( function() {
+	'use strict';
+
+	/**
+	 * Providing messages using two message providers
+	 * @class util.CombiningMessageProvider
+	 * @licence GNU GPL v2+
+	 * @author H. Snater < mediawiki@snater.com >
+	 * @author Adrian Heine < adrian.heine@wikimedia.de >
+	 *
+	 * @constructor
+	 *
+	 * @param {util.MessageProvider} a
+	 * @param {util.MessageProvider} b
+	 */
+	function CombiningMessageProvider( a, b ) {
```

Comment:
Sorry, maybe it is just me who does not really like short variable names. Looking at the constructor arguments, I was wondering how both `MessageProvider`s "interact" as `a` and `b` did not really give me a hint. Maybe the parameters should be named `primaryMessageProvider` (or `preferredMessageProvider`) and `secondaryMessageProvider` (or `alternativeMessageProvider`) - just an idea. If the parameter was an array, I probably would have guessed the mechanism right away.